diff --git a/src/test/regress/sql/limit.sql b/src/test/regress/sql/limit.sql
index 603910fe6d..0cfef4c77b 100644
--- a/src/test/regress/sql/limit.sql
+++ b/src/test/regress/sql/limit.sql
@@ -39,7 +39,7 @@ select * from int8_tbl offset (case when random() < 0.5 then null::bigint end);
 -- Test assorted cases involving backwards fetch from a LIMIT plan node
 begin;
 
-declare c1 cursor for select * from int8_tbl limit 10;
+declare c1 scroll cursor for select * from int8_tbl limit 10;
 fetch all in c1;
 fetch 1 in c1;
 fetch backward 1 in c1;
@@ -47,7 +47,7 @@ fetch backward all in c1;
 fetch backward 1 in c1;
 fetch all in c1;
 
-declare c2 cursor for select * from int8_tbl limit 3;
+declare c2 scroll cursor for select * from int8_tbl limit 3;
 fetch all in c2;
 fetch 1 in c2;
 fetch backward 1 in c2;
@@ -55,7 +55,7 @@ fetch backward all in c2;
 fetch backward 1 in c2;
 fetch all in c2;
 
-declare c3 cursor for select * from int8_tbl offset 3;
+declare c3 scroll cursor for select * from int8_tbl offset 3;
 fetch all in c3;
 fetch 1 in c3;
 fetch backward 1 in c3;
@@ -63,7 +63,7 @@ fetch backward all in c3;
 fetch backward 1 in c3;
 fetch all in c3;
 
-declare c4 cursor for select * from int8_tbl offset 10;
+declare c4 scroll cursor for select * from int8_tbl offset 10;
 fetch all in c4;
 fetch 1 in c4;
 fetch backward 1 in c4;
@@ -71,7 +71,7 @@ fetch backward all in c4;
 fetch backward 1 in c4;
 fetch all in c4;
 
-declare c5 cursor for select * from int8_tbl order by q1 fetch first 2 rows with ties;
+declare c5 scroll cursor for select * from int8_tbl order by q1 fetch first 2 rows with ties;
 fetch all in c5;
 fetch 1 in c5;
 fetch backward 1 in c5;
