diff --git a/src/test/perl/PostgreSQL/Test/Cluster.pm b/src/test/perl/PostgreSQL/Test/Cluster.pm
index b95ba721f71..ea04f1c0b1b 100644
--- a/src/test/perl/PostgreSQL/Test/Cluster.pm
+++ b/src/test/perl/PostgreSQL/Test/Cluster.pm
@@ -1060,6 +1060,15 @@ sub start
 	}

 	$self->_update_pid(1);
+
+	if ((! -e $self->data_dir . '/standby.signal') && (! -e $self->data_dir . '/recovery.signal'))
+	{
+		$self->safe_psql(
+			'template1', qq{CREATE EXTENSION IF NOT EXISTS orioledb;});
+		$self->safe_psql(
+			'postgres', qq{CREATE EXTENSION IF NOT EXISTS orioledb;});
+	}
+
 	return 1;
 }

